ironfish/src/network/peerNetwork.ts

```diff
@@ -725,6 +729,8 @@ export class PeerNetwork {
       return true
     }
 
+    this.onTransactionSeen.emit(transaction, received)
```

Comment:
This is creating a bug, to move this here, you need to refactor the above logic.